### PR TITLE
feat: Support Code Security Configurations API

### PR DESCRIPTION
**Pull request by maditya**
This PR implements Code Security Configurations APIs as defined at https://docs.github.com/en/rest/code-security/configurations?apiVersion=2022-11-28

Fixes https://github.com/google/go-github/issues/3211

### CODE REVIEW

File: github/orgs_codesecurity_configurations.go
Comment (gmlewis) [inline github/orgs_codesecurity_configurations.go:1]:
```suggestion
// Copyright 2024 The go-github AUTHORS. All rights reserved.
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package github
```

File: github/orgs_codesecurity_configurations_test.go
Comment (gmlewis) [inline github/orgs_codesecurity_configurations_test.go:1]:
```suggestion
// Copyright 2024 The go-github AUTHORS. All rights reserved.
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package github
```

File: github/orgs_codesecurity_configurations.go
Comment (gmlewis) [inline github/orgs_codesecurity_configurations.go:11]:
```suggestion
	LabeledRunners *bool `json:"labeled_runners,omitempty"`
```

File: github/orgs_codesecurity_configurations.go
Comment (gmlewis) [inline github/orgs_codesecurity_configurations.go:266]:
```suggestion
func (s *OrganizationsService) GetCodeSecurityConfigurationForRepository(ctx context.Context, org, repo string) (*RepositoryCodeSecurityConfiguration, *Response, error) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @maditya!
Just a few tweaks, and please remember to run `go generate ./...` as mentioned in step 4 of CONTRIBUTING.md and push (not force-push) the changes to this PR.
We then should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Comment by maditya** [comment]
@gmlewis Thanks!
I tried and fixed `go generate ./...` which updated a bunch of other files.
But `metadata` script is failing -

```
ambiguous operation "GET /user/{user_id}" could match any of: [GET /user/{account_id} GET /user/{user_id}]
github/github.go:8: running "../script/metadata.sh": exit status 1
```
Any idea how to address it? Doesn't seem related to the new API.

**Comment by gmlewis** [comment]
> Any idea how to address it?

Hmmm... a lot more files changed than I expected.
It looks like you may have locally updated the OpenAPI specs, which is typically done by a maintainer of the repo.

We have two options:
1. revert the changes to all the files except your original two, then run `go generate ./...` 
2. wait for another PR to update all the OpenAPI stuff and fix all the errors you are seeing

I think option 1 is probably better since option 2 may take a while, and I'm currently monitoring a hurricane locally.

**Comment by maditya** [comment]
>  revert the changes to all the files except your original two, then run go generate ./...

Done. Note that `go generate ./...` updates some other files, namely `github/admin_users.go`, `github/github-accessors.go` and `github/github-accessors_test.go` and will fail until `openapi_operations.yaml` is updated. I get these errors: 
```
could not find operation "GET /orgs/{org}/code-security/configurations" in openapi_operations.yaml
could not find operation "POST /orgs/{org}/code-security/configurations" in openapi_operations.yaml
could not find operation "GET /orgs/{org}/code-security/configurations/defaults" in openapi_operations.yaml
could not find operation "DELETE /orgs/{org}/code-security/configurations/detach" in openapi_operations.yaml
could not find operation "GET /orgs/{org}/code-security/configurations/{configuration_id}" in openapi_operations.yaml
could not find operation "PATCH /orgs/{org}/code-security/configurations/{configuration_id}" in openapi_operations.yaml
could not find operation "DELETE /orgs/{org}/code-security/configurations/{configuration_id}" in openapi_operations.yaml
could not find operation "POST /orgs/{org}/code-security/configurations/{configuration_id}/attach" in openapi_operations.yaml
could not find operation "PUT /orgs/{org}/code-security/configurations/{configuration_id}/defaults" in openapi_operations.yaml
could not find operation "GET /orgs/{org}/code-security/configurations/{configuration_id}/repositories" in openapi_operations.yaml
could not find operation "GET /repos/{owner}/{repo}/code-security-configuration" in openapi_operations.yaml
github/github.go:8: running "../script/metadata.sh": exit status 1
```

**Comment by gmlewis** [comment]
> Note that `go generate ./...` updates some other files

That is really strange. 
`go generate ./...` should only update three files (as needed):

* github/github-accessors.go 
* github/github-accessors_test.go
* github/github-stringify_test.go

I'm going to make a new branch locally with just your changes to your two files and run `go generate ./...` and see what happens for me and report back.

**Comment by gmlewis** [comment]
> I'm going to make a new branch locally with just your changes to your two files and run `go generate ./...` and see what happens for me and report back.

Ah, I see what is going on now.  You are adding support for new operations not yet defined in our local OpenAPI yaml.

This section in CONTRIBUTING.md is pertinent to you:

> ### openapi_operations.yaml
> 
> Most contributors won't need to interact with `openapi_operations.yaml`, but it
> may be useful to know what it is. Its sections are:
> ...

So you can either attempt to update it yourself by running `script/metadata.sh update-openapi` and then work out any issues you find, or I can do that in a separate PR.  Which would you prefer?

**Comment by gmlewis** [comment]
EDIT: I've gone ahead and created #3322 which you should be able to merge into your PR (from master once I merge it), which should make the rest of this PR easier to finish.

It looks like (in addition to the code generation), the tests need to have more coverage by using existing test helper methods like `testBadOptions` and `testNewRequestAndDoFailure`. You can see these in the GitHub UI.

**Review by gmlewis — APPROVED** [review]
Thank you, @maditya !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by gmlewis** [comment]
Thank you, @tomfeigin !
Merging.

**Pull request merged**
